apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.SignedPostPolicy.cs

```diff
@@ -0,0 +1,83 @@
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.Cloud.Storage.V1
+{
+    public sealed partial class UrlSigner
+    {
+        /// <summary>
+        /// The result of signing a <see cref="PostPolicy"/>
+        /// containing the fields that should be including when
+        /// making the post request.
+        /// </summary>
+        public sealed class SignedPostPolicy
+        {
+            private const string SignatureElementName = "x-goog-signature";
+            private const string PolicyElementName = "policy";
+
+            internal SignedPostPolicy(PostPolicy source, string encodedPolicy, string signature, Uri url)
+            {
+                Source = GaxPreconditions.CheckNotNull(source, nameof(source));
+                PostUrl = GaxPreconditions.CheckNotNull(url, nameof(url));
+                EncodedPolicy = GaxPreconditions.CheckNotNull(encodedPolicy, nameof(encodedPolicy));
+                Signature = GaxPreconditions.CheckNotNull(signature, nameof(signature));
+            }
+
+            internal PostPolicy Source { get; }
+
+            internal string EncodedPolicy { get; }
+
+            internal string Signature { get; }
+
+            /// <summary>
+            /// The URL where the post request enforced by this signed policy should be made.
+            /// </summary>
+            public Uri PostUrl { get; }
+
+            /// <summary>
+            /// Collection of fields as defined by this policy.
+            /// Post requests enforced by this policy should set, for each
+            /// pair in this sequence, the given element to the exact given value.
+            /// </summary>
+            public IEnumerable<(string element, string value)> Fields =>
+                Source.Fields.Concat(new (string element, string value)[] 
+                {
+                    (SignatureElementName, Signature),
+                    (PolicyElementName, EncodedPolicy)
+                });
+
+            /// <summary>
+            /// Collection of start with conditions as defined by this policy.
+            /// Post requests enforced by this policy should set, for each
+            /// pair in this sequence, the given element to a value starting with the given prefix.
+            /// </summary>
+            public IEnumerable<(string element, string prefix)> StartWithConditions =>
+                Source.StartWithConditions;
+
+            /// <summary>
+            /// Collection of in range conditions as defined by this policy.
+            /// Post requests enforced by this policy should set, for each
+            /// tuple in this sequence, the given element to a value that's in the range defined by
+            /// min and max values.
+            /// </summary>
+            public IEnumerable<(string element, object min, object max)> InRangeCondition =>
```

Comment:
Same here: Could you provide more information on what this helper does?

Reply:
Same as above.